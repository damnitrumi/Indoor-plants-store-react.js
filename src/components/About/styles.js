import styled, { css } from "styled-components";
import { Container as SectionWrapper } from "../SectionWrapper/styles";
import { Container as Text } from "../Text/styles";

export const Container = styled.div`
  ${() => css`
    counter-reset: TextCounter;
    ${SectionWrapper} {
      display: flex;
      flex-direction: column;
      gap: 180px;
    }
  `}
`;
export const AboutContainer = styled.div`
  ${({ theme }) => css`
    ${Text}::before {
      content: counter(TextCounter) ".";
      counter-increment: TextCounter;
      font-size: ${theme.font.sizes.large};
      font-family: ${theme.font.family.primary};
      margin-right: ${theme.spacings.small};
      color: ${theme.colors.lightGreen};
    }

    padding: ${theme.spacings.small} 0;
    /* margin-top: 180px; */
    overflow-y: auto;
  `}
`;
