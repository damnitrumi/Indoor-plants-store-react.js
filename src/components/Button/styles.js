import styled, { css } from "styled-components";

export const Container = styled.button`
  ${({ theme }) => css`
    background: transparent;
    border: none;
    cursor: pointer;
    padding: ${theme.spacings.tiny};
    color: white;
    position: relative;
  `}
`;
