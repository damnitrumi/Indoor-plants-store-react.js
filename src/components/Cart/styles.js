import styled, { css } from "styled-components";

export const Container = styled.div`
  ${() => css`
    width: 360px;
  `}
`;

export const ProductContainer = styled.div`
  ${() => css``}
`;

export const ProductImageContainer = styled.div`
  ${() => css``}
`;
export const ProductData = styled.div`
  ${() => css``}
`;

export const NameContainer = styled.div`
  ${() => css`
    display: flex;
    justify-content: space-between;
  `}
`;

export const ProductOptionsContainer = styled.div`
  ${() => css``}
`;

export const QuantityContainer = styled.div`
  ${() => css``}
`;
