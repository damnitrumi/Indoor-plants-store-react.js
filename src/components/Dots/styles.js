import styled, { css } from "styled-components";

export const Container = styled.span`
  ${({ theme, current, index }) => css`
    display: inline-block;
    border-radius: 50%;
    height: ${theme.spacings.tiny};
    width: ${theme.spacings.tiny};
    background-color: ${current == index
      ? theme.colors.grey
      : theme.colors.mediumGrey};
  `}
`;
