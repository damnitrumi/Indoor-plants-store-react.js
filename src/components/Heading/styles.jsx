import styled, { css } from "styled-components";

export const Container = styled.h1`
  ${({ theme }) => css`
    font-family: ${theme.font.family.primary};
    color: ${theme.colors.darkGreen};
    margin-bottom: ${theme.spacings.medium};
    font-size: ${theme.font.sizes.large};
    text-align: center;
  `}
`;
