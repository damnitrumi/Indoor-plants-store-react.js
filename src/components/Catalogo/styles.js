import styled, { css } from "styled-components";
import { Container as SectionWrapper } from "../SectionWrapper/styles";
import { Container as ImageContainer } from "../ImageContainer/styles";

// export const SectionContainer = styled.div`
//   ${() => css`
//     display: flex;
//     flex-direction: column;
//     height: 100%;
//   `}
// `;

export const Container = styled.div`
  ${({ visible }) => css`
    height: 100vh;
    display: flex;
    flex-direction: column;
    transition: transform 0.5s;
    transform: translateX(-100%);
    position: absolute;
    width: 360px;
    ${visible &&
    css`
      transform: translateX(0);
    `}
    ${SectionWrapper} {
      flex: 1;
      display: flex;
      flex-direction: column;
    }
  `}
`;

export const SlidesContainer = styled.div`
  ${() => css`
    /* ${ImageContainer} {
      height: 460px;
    } */
  `}
`;

export const SliderContainer = styled.div`
  ${() => css`
    flex: 1;
  `}
`;

export const BgContainer = styled.span`
  ${({ theme }) => css`
    display: block;
    background-color: ${theme.colors.lightGrey};
    width: 230px;
    height: 300px;
    border-radius: ${theme.spacings.medium};
    position: absolute;
    z-index: -1;
    top: 40px;
  `}
`;

export const DotsContainer = styled.div`
  ${() => css`
    display: flex;
    justify-content: center;
    gap: 3px;
    position: absolute;
    left: 150px;
    bottom: 290px;
  `}
`;
