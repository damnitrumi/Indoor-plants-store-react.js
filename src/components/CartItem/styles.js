import styled, { css } from "styled-components";

import { Container as Heading } from "../Heading/styles";
import { Container as Button } from "../Button/styles";

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    margin-top: ${theme.spacings.small};
    ${Heading} {
      margin-bottom: 0;
      font-size: 1.4rem;
      font-family: ${theme.font.family.secondary};
    }
    ${Button} {
      color: initial;
    }
  `}
`;

export const ProductImageContainer = styled.div`
  ${() => css`
    flex: 1;
  `}
`;
export const ProductData = styled.div`
  ${({ theme }) => css`
    flex: 2;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding-right: ${theme.spacings.large};
  `}
`;

export const NameContainer = styled.div`
  ${() => css`
    display: flex;
    justify-content: space-between;
  `}
`;

export const ProductOptionsContainer = styled.div`
  ${() => css``}
`;

export const QuantityContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    font-size: ${theme.font.sizes.medium};
  `}
`;
export const Quantity = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    border: 1px solid rgba(0, 0, 0, 0.3);
    ${Button}:first-of-type {
      border-right: 1px solid rgba(0, 0, 0, 0.3);
    }

    ${Button}:last-of-type {
      border-left: 1px solid rgba(0, 0, 0, 0.3);
    }

    span {
      padding: 0 ${theme.spacings.tiny};
    }
  `}
`;
