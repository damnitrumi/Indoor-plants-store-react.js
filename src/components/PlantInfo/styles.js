import styled, { css } from "styled-components";
import { Container as Heading } from "../Heading/styles";
import { Container as Button } from "../Button/styles";
import { Container as Text } from "../Text/styles";

const activeBtn = () => css`
  display: block;
`;

export const Container = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.grey};
    width: 100%;
    flex: 0.7;
    border-radius: ${theme.spacings.large} ${theme.spacings.large} 0 0;
    padding: ${theme.spacings.large} ${theme.spacings.xlarge};
    display: flex;
    flex-direction: column;
    ${Heading} {
      color: white;
      margin-bottom: 0;
      font-size: ${theme.font.sizes.small};
    }
    > ${Button} {
      text-align: center;
      background-color: ${theme.colors.lightGreen};
      display: block;
      width: 85%;
      margin: 0 auto;
      border-radius: 5px;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;

export const NameContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    margin-bottom: ${theme.spacings.medium};
  `}
`;

export const TabsWrapper = styled.div`
  ${() => css`
    flex: 1;
    display: flex;
    flex-direction: column;
  `}
`;

export const TabsContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    padding: 0 ${theme.spacings.tiny};
    position: relative;
  `}
`;

export const Tab = styled.div`
  ${({ theme }) => css`
    flex: 1;
    display: flex;
    flex-direction: column;
    ${Text} {
      padding: ${theme.spacings.tiny};
      color: ${theme.colors.mediumGrey};
      font-size: 1.3rem;
      flex: 1;
      max-height: 60px;
      overflow: auto;
    }
  `}
`;

export const PlantBtn = styled(Button)`
  ${({ theme, idBtn, tab }) => css`
    position: relative;
    letter-spacing: 0.05rem;

    &::after {
      display: none;
      content: "";
      background-color: ${theme.colors.mediumGreen};
      width: 100%;
      height: 3px;
      position: absolute;
      left: -1%;
      ${idBtn == tab && activeBtn()}
    }
  `}
`;
