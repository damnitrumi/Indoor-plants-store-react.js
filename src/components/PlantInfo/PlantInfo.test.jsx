import { screen } from "@testing-library/react";
import { renderTheme } from "../../styles/render-theme";
import { PlantInfo } from ".";
import plantsMock from "../utils/plantsMock";
import userEvent from "@testing-library/user-event";

describe("<PlantInfo />", () => {
  it("should render PlantInfo", () => {
    renderTheme(<PlantInfo plant={plantsMock[0]} />);

    const firstHeading = screen.getByRole("heading", { name: "Cacto Plant 1" });
    const secondHeading = screen.getByRole("heading", {
      name: "R$ 21.00",
    });
    const headings = screen.getAllByRole("heading");

    expect(firstHeading).toBeInTheDocument();
    expect(secondHeading).toBeInTheDocument();
    expect(headings).toHaveLength(2);
  });

  it("should render the correct tab", () => {
    renderTheme(<PlantInfo plant={plantsMock[0]} />);

    const buttons = screen.getAllByRole("button");

    userEvent.click(buttons[0]);

    const details = screen.getByText("Cactus are succulent perennial plants 1");

    expect(details).toBeInTheDocument();
  });

  it("should render the correct tab", () => {
    renderTheme(<PlantInfo plant={plantsMock[0]} />);

    const buttons = screen.getAllByRole("button");

    userEvent.click(buttons[1]);

    const reviews = screen.getByText(
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc lobortis aliquam elit. Vestibulum eget metus 1"
    );

    expect(reviews).toBeInTheDocument();
  });

  it("should render the correct tab", () => {
    renderTheme(<PlantInfo plant={plantsMock[0]} />);

    const buttons = screen.getAllByRole("button");

    userEvent.click(buttons[2]);

    const about = screen.getByText(
      "Nulla eu massa facilisis, pulvinar nulla at, varius mauris. Fusce id fringilla dolor, ac auctor. 1"
    );

    expect(about).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = renderTheme(<PlantInfo plant={plantsMock[0]} />);

    expect(container).toMatchSnapshot();
  });
});
