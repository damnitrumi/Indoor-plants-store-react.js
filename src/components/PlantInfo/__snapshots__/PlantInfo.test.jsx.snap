// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlantInfo /> should match snapshot 1`] = `
.c3 {
  font-family: 'Aclonica',sans-serif;
  color: #213e3a;
  margin-bottom: 2.4rem;
  font-size: 3.2rem;
  text-align: center;
}

.c7 {
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0.8rem;
  color: white;
}

.c12 {
  margin-bottom: 2.4rem;
  font-size: 1.6rem;
  font-family: 'Montserrat',sans-serif;
  text-align: justify;
}

.c0 {
  background-color: #303030;
  width: 100%;
  -webkit-flex: 0.7;
  -ms-flex: 0.7;
  flex: 0.7;
  border-radius: 3.2rem 3.2rem 0 0;
  padding: 3.2rem 4.0rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .c2 {
  color: white;
  margin-bottom: 0;
  font-size: 1.6rem;
}

.c0 > .c6 {
  text-align: center;
  background-color: #009d84;
  display: block;
  width: 85%;
  margin: 0 auto;
  border-radius: 5px;
  font-size: 1.6rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 2.4rem;
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 0.8rem;
  position: relative;
}

.c10 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c10 .c11 {
  padding: 0.8rem;
  color: #c7c8c8;
  font-size: 1.3rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  max-height: 60px;
  overflow: auto;
}

.c8 {
  position: relative;
  -webkit-letter-spacing: 0.05rem;
  -moz-letter-spacing: 0.05rem;
  -ms-letter-spacing: 0.05rem;
  letter-spacing: 0.05rem;
}

.c8::after {
  display: none;
  content: "";
  background-color: #1b6359;
  width: 100%;
  height: 3px;
  position: absolute;
  left: -1%;
  display: block;
}

.c9 {
  position: relative;
  -webkit-letter-spacing: 0.05rem;
  -moz-letter-spacing: 0.05rem;
  -ms-letter-spacing: 0.05rem;
  letter-spacing: 0.05rem;
}

.c9::after {
  display: none;
  content: "";
  background-color: #1b6359;
  width: 100%;
  height: 3px;
  position: absolute;
  left: -1%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        class="c2 c3"
      >
        Cacto Plant 1
      </h1>
      <h1
        class="c2 c3"
      >
        R$ 21.00
      </h1>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <button
          class="c6 c7 c8"
        >
          Details
        </button>
        <button
          class="c6 c7 c9"
        >
          Reviews
        </button>
        <button
          class="c6 c7 c9"
        >
          About
        </button>
      </div>
      <div
        class="c10"
      >
        <p
          class="c11 c12"
        >
          Cactus are succulent perennial plants 1
        </p>
      </div>
    </div>
    <button
      class="c6 c7"
    >
      Add To Cart
    </button>
  </div>
</div>
`;
